lib/pbench/server/api/resources/datasets_list.py

```diff
@@ -129,8 +146,11 @@ def _get(
         """
         json = params.query
 
+        # Get the authenticated user ID, if any
+        user_id = Auth.get_current_user_id()
```

Comment:
This will for an additional lookup in the database which is not necessary unless `filter` is requested.  Can this be moved to after that check on line 162?

Reply:
It is just used in the `user` key processing, but I didn't want to call it each time. Putting it inside the `if filter` block is fine; I'd rather not move it further in.

Reply:
I can see how we might not want it inside the loop; on the other hand, it's annoying to make that call if we're not going to use the value.

You could consider initializing `user_id` to `None` at the top and then calling `get_current_user_id()` in the `native_key == Metadata.USER` block iff `user_id` is still `None` when you get there -- that way we would call it at most once per filtration, and zero times if none of the filters reference the `user` space.